app/controllers/cloud_object_store_container_controller.rb

```diff
@@ -6,17 +6,17 @@ class CloudObjectStoreContainerController < ApplicationController
 
   include Mixins::GenericListMixin
   include Mixins::GenericShowMixin
+  include Mixins::GenericButtonMixin
   include Mixins::GenericSessionMixin
 
   def breadcrumb_name(_model)
     ui_lookup(:tables => "cloud_object_store_container")
   end
 
-  # handle buttons pressed on the button bar
   def button
```

Comment:
Yeah, the indirection can be annoying, but it can be overcome with good naming and there are a lot of benefits.

Giving things an intention-revealing name helps people down the road. For instance, it might be easy to read `@display = params[:display]` but why on earth is that being done? In some places there are comments explaining why, but then that might as well be in a method of that name.

So, for instance I have a `restore_edit_for_search` method where the name is longer than the code. It might seem like a waste, but now the method says why it's being done. Plus, of course, if the implementation of restoring edit for search ever changes there's one place to do it.

Even more useful to this refactoring is that extracting methods is helping me find what's common even when it's buried in the 3rd level of nested `if` statements.

Also, some of this is just a transition until I find better ways of simplifying. I will probably get rid of some of these methods.